framework/Distributions.py

```diff
@@ -2441,6 +2443,116 @@ def rvs(self):
 
 DistributionsCollection.addSub(Custom1D.getInputSpecification())
 
+class LogUniform(Distribution):
+  """
+    Log Uniform univariate distribution
+    If x~LogUnif(a,b) then log(x)~Unif(log(a),log(b))
+  """
+  @classmethod
+  def getInputSpecification(cls):
+    """
+      Method to get a reference to a class that specifies the input data for
+      class cls.
+      @ In, cls, the class for which we are retrieving the specification
+      @ Out, inputSpecification, InputData.ParameterInput, class to use for
+        specifying input of cls.
+    """
+    inputSpecification = super(LogUniform, cls).getInputSpecification()
+    inputSpecification.addSub(InputData.parameterInputFactory("lowerBound", contentType=InputData.FloatType))
+    inputSpecification.addSub(InputData.parameterInputFactory("upperBound", contentType=InputData.FloatType))
+    inputSpecification.addSub(InputData.parameterInputFactory("base"      , contentType=InputData.StringType))
+
+    return inputSpecification
+
+  def __init__(self):
+    """
+      Constructor
+      @ In, None
+      @ Out, None
+    """
+    Distribution.__init__(self)
+    self.upperBound = None
+    self.lowerBound = None
+    self.base       = None
+
+  def _handleInput(self, paramInput):
+    """
+      Function to handle the common parts of the distribution parameter input.
+      @ In, paramInput, ParameterInput, the already parsed input.
+      @ Out, None
+    """
+    self.lowerBound = paramInput.findFirst('lowerBound').value
+    if self.lowerBound is None:
+      self.raiseAnError(IOError,' lowerBound parameter is needed for LogUniform distribution')
+
+    self.upperBound = paramInput.findFirst('upperBound').value
+    if self.upperBound is None:
```

Comment:
I think this checks are not needed since you have the inputChecker. If it does not find the parameters, It errors out on its own.